tx-pool/src/component/pending.rs

```diff
@@ -70,6 +70,13 @@ impl PendingQueue {
             }
         }
     }
+
+    pub(crate) fn drain(&mut self) -> Vec<TransactionView> {
+        self.inner
+            .values()
+            .map(|entry| entry.transaction().clone())
+            .collect()
+    }
```

Comment:
### Request Changes

This method is incorrect: it should clear `inner`.

p.s. Pending since I'm waiting more request changes and I will fix them in batch.